Mathlib/Analysis/CauchyBound.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Analysis.Normed.Field.Basic
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset NNReal
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
+  sup (range p.natDegree) (‖p.coeff ·‖₊) / ‖p.leadingCoeff‖₊
+
+lemma NNReal.geom_sum {x : ℝ≥0} (h : 1 < x) (n : ℕ) :
+    ∑ i ∈ Finset.range n, x ^ i = (x ^ n - 1) / (x - 1) := by
+  apply eq_div_of_mul_eq (tsub_pos_of_lt h).ne'
+  apply eq_tsub_of_add_eq
+  convert geom_sum_mul_add (x - 1) n <;>
+  rw [tsub_add_cancel_of_le h.le]
+
+theorem norm_lt_cauchyBound_add_one_of_isRoot (p : K[X]) (hp : p ≠ 0) (a : K) (h : p.IsRoot a) :
```

Comment:
`Polynomial.IsRoot.norm_lt_cauchyBound_add_one`? Also, I think that the definition of `cauchyBound` should include this `+ 1`. BTW, do I remember correctly that it works for `max 1 (cauchyBound p)`?

Also, please add a version about `Polynomial.aeval`.

Reply:
Regarding `aeval` - I discussed this a bit on Zulip at https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/NormedAlgebra.20with.20ring, I'm not sure how to state this in the correct generality for an algebra

Reply:
There is Lagrange's bound, which is $\max(1, \sum_{i=0}^{n-1} \left| \frac{a_i}a_n \right|)$ (compared to Cauchy's bound, $1 + \max_{i=0}^{n-1}{\left|\frac{a_i}{a_n} \right|}$.